picasso/src/main/java/com/squareup/picasso3/StatsSnapshot.java

```diff
@@ -63,47 +64,69 @@
 
   /** Prints out this {@link StatsSnapshot} into log. */
   @SuppressWarnings("UnusedDeclaration") public void dump() {
-    StringWriter logWriter = new StringWriter();
-    dump(new PrintWriter(logWriter));
-    Log.i(TAG, logWriter.toString());
+    Buffer buffer = new Buffer();
+    try {
+      dump(buffer);
+    } catch (IOException e) {
```

Comment:
propagate the IOException to the caller?

Reply:
There's no I/O happening here, writing to the buffer. It's an impossible case and would force the caller to assert that it can't happen.

Reply:
got it.  didn't realize Buffer's implementation of BufferSink.writeUtf8 doesn't declare the IOException in its signature.  nvm.